tensorflow/core/kernels/data/zip_dataset_op.cc

```diff
@@ -136,21 +136,32 @@ class ZipDatasetOp::Dataset : public DatasetBase {
       }
       out_tensors->clear();
       out_tensors->reserve(dataset()->output_dtypes().size());
+      Status status = Status::OK();
+      *end_of_sequence = false;
       for (const auto& input_impl : input_impls_) {
         std::vector<Tensor> input_tensors;
-        TF_RETURN_IF_ERROR(
-            input_impl->GetNext(ctx, &input_tensors, end_of_sequence));
+        bool component_end_of_sequence = false;
+        Status component_status = input_impl->GetNext(ctx, &input_tensors, &component_end_of_sequence);
+        *end_of_sequence |= component_end_of_sequence;
+        status.Update(component_status);
```

Comment:
This could be further simplified to:

```
status.Update(input_impl->GetNext(ctx, &input_tensors, &component_end_of_sequence);
*end_of_sequence |= component_end_of_sequence;
```

Reply:
Thanks @jsimsa! This looks even better.